internal/util/funcutil/parallel.go

```diff
@@ -31,13 +32,22 @@ func GetFunctionName(i interface{}) string {
 	return runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
 }
 
-// ProcessFuncParallel process function in parallel.
+type TaskFunc func() error
+type ProcessFunc func(idx int) error
+type DataProcessFunc func(data interface{}) error
+
+// ProcessFuncParallel processes function in parallel.
 //
 // ProcessFuncParallel waits for all goroutines done if no errors occur.
 // If some goroutines return error, ProcessFuncParallel cancels other goroutines as soon as possible and wait
 // for all other goroutines done, and returns the first error occurs.
 // Reference: https://stackoverflow.com/questions/40809504/idiomatic-goroutine-termination-and-error-handling
-func ProcessFuncParallel(total, maxParallel int, f func(idx int) error, fname string) error {
+func ProcessFuncParallel(total, maxParallel int, f ProcessFunc, fname string) error {
```

Comment:
https://github.com/git-hulk/routines
https://github.com/goinggo/workpool/blob/master/workpool.go
recommend to use some routine pool(not sure the one listed is the best choise), which could be more elegant and